File: src/backend/StackSift.Serilog.Sink/Internal/LevelMapping.cs
using Serilog.Events;

namespace StackSift.Serilog.Sink.Internal;

internal static class LevelMapping
{
    public static string ToStackSift(LogEventLevel level) => level switch
    {
        LogEventLevel.Verbose => "Trace",
        LogEventLevel.Debug => "Debug",
        LogEventLevel.Information => "Info",
        LogEventLevel.Warning => "Warning",
        LogEventLevel.Error => "Error",
        LogEventLevel.Fatal => "Critical",
        _ => "Info",
    };
}

File: sdks/winston-transport/src/levelMapping.ts
const NPM_TO_STACKSIFT: Record<string, string> = {
  silly: 'Trace',
  trace: 'Trace',
  verbose: 'Debug',
  debug: 'Debug',
  info: 'Info',
  http: 'Info',
  warn: 'Warning',
  warning: 'Warning',
  error: 'Error',
  fatal: 'Critical',
  critical: 'Critical',
};

export function mapLevel(level: string): string {
  const key = (level ?? '').toLowerCase();
  return NPM_TO_STACKSIFT[key] ?? 'Info';
}
